YAXLib/Custom/ITypeContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
```

Comment:
Provides all `Type`-specific information.
Newly added:
* `IEnumerable<MemberContext> GetFieldsForSerialization()`: For processing fields one-by-one
* `IEnumerable<MemberContext> GetFieldsForDeserialization()`
* `XElement Serialize(object obj, SerializerOptions options = null)`: All members of a type can be serialized without using the current `ICustomSerializer` or `IKnownType`
* `object Deserialize(XElement element, SerializerOptions options = null)`: All members of a type can be deserialized without using the current `ICustomSerializer` or `IKnownType`